qiskit/opflow/gradients/circuit_gradients/lin_comb.py

```diff
@@ -665,11 +692,45 @@ def _gradient_states(
                         param_expression = gate.params[idx]
 
                         if isinstance(meas_op, OperatorBase):
-                            state = meas_op @ state
-                        elif meas_op is True:
-                            state = ListOp(
-                                [state], combo_fn=partial(self._grad_combo_fn, state_op=state_op)
-                            )
+                            state = StateFn(meas_op, is_measurement=True) @ state
+
+                        else:
```

Comment:
Is it correct that the different types of `aux_meas_op`s are only supported if no `meas_op` is provided? If yes then we should raise an error if someone tries passing both 🤔

Reply:
That's a good point. 
If a `meas_op` is given and an `aux_meas_op` is given then the `_prepare_operator` function combines them directly. If no `meas_op` is given this cannot be done. Therefore, the `aux_meas_op` is propagated further. I agree that this is not ideal. 
We could

1. Raise a warning if both are giving at this point in the code.
2. Move the correct combination of `meas_op` and `aux_meas_op` to this part of the code. 
3. Instead of propagating through the `aux_meas_op` set the `meas_op` to be the `aux_meas_op`. Although, I would like to avoid mixing these two concepts.

I personally prefer option 2. Let me know what you think.